cpp/src/gandiva/tests/projector_test.cc

```diff
@@ -1606,4 +1606,47 @@ TEST_F(TestProjector, TestCastNullableIntYearInterval) {
   EXPECT_ARROW_ARRAY_EQUALS(out_int64, outputs.at(1));
 }
 
+TEST_F(TestProjector, TestInstr) {
+  // schema for input fields
+  auto field0 = field("f0", arrow::utf8());
+  auto field1 = field("f2", arrow::utf8());
+  auto schema = arrow::schema({field0, field1});
+
+  // output fields
+  auto output_instr = field("out_instr", int32());
+
+
+  // Build expression
+  auto instr_expr = TreeExprBuilder::MakeExpression("instr",
+                                                    {field0, field1}, output_instr);
+
+
+  std::shared_ptr<Projector> projector;
+  auto status =
+      Projector::Make(schema, {instr_expr}, TestConfiguration(), &projector);
+  EXPECT_TRUE(status.ok());
+
+  // Create a row-batch with some sample data
+  int num_records = 4;
+  auto array0 = MakeArrowArrayUtf8(
+      {"hello world!", "apple, banana, mango", "", "open the door"},
+      {true, true, true, true});
+  auto array1 = MakeArrowArrayUtf8(
+      {"world", "mango", "mango", ""},
```

Comment:
What is the expected output if the substr is not null, but the length is 0? Is this matching at index 1 or not matching?

The Hive documentation is not clear on this. Can you test how the Hive function behaves? My testing shows that an empty substring (not-null, but a substring of "") returns 1 for the INSTR(column, '')

Reply:
Yest, I tested it here too, and it now returns 1